### PR TITLE
Add generated field to listunspent output

### LINKED ISSUE #2446: Add field indicating coinbase tx in output of listunspent
This comes from a user support request. Would be useful, in particular for large wallets, to not have to use gettransaction to determine whether an input is a coinbase reward. Suggestion is to add a field such as `"generated": true`

Unless I'm overlooking some existing way to find out from the output of listunspent?

### PR DESCRIPTION
**Pull request by bitcartel**
Closes #2446

### CODE REVIEW

**Comment by bitcartel** [comment]
Note: I thought #2446 was in the 1.0.11 milestone but it seems I was mistaken, although it's related to #2449 which is in the milestone.  This PR can help third parties such as wallets and exchanges identify coinbase utxos which must be shielded.

File: qa/rpc-tests/wallet.py [hunk 1]
```diff
@@ -72,6 +72,36 @@ def run_test (self):
...
         assert_equal(len(node0utxos), 3)
 
+        # Check 'generated' field of listunspent
+        #
+        # Node 0: has one coinbase utxo and two regular utxos
+        generatedCounter = 0
+        for utxo in node0utxos:
+            if utxo["generated"] is True:
+                generatedCounter = generatedCounter + 1
+        assert(generatedCounter == 1)
```
Comment (daira) [inline qa/rpc-tests/wallet.py]: You can write this as ``assert_equal(sum(int(uxto["generated"] is True) for uxto in node0utxos), 1)``, and similarly below.

Comment (daira) [inline qa/rpc-tests/wallet.py]: (See https://stackoverflow.com/a/2900111/393146. )

**Comment by daira** [comment]
I think it was an oversight that #2446 was not proposed for the milestone. In any case I've added it since the work on the PR and my review is a sunk cost now, and an exchange was requesting it.

File: qa/rpc-tests/wallet.py (hunk 1 above)
Comment (bitcartel) [inline qa/rpc-tests/wallet.py]: I would prefer to use something like 'coinbase' (over 'isCoinbase' which is camel case) but the ticket calls for 'generated' and it's already being used in one location here: https://github.com/zcash/zcash/blob/master/src/wallet/rpcwallet.cpp#L80

Comment (bitcartel) [inline qa/rpc-tests/wallet.py]: Updated test to be more python-esque!

**Comment by bitcartel** [comment]
@zkbot try

**Comment by str4d** [comment]
+1 on using "generated" in line with other usage in the transparent APIs.

**Comment by daira** [comment]
ut(ACK+cov)

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**